GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -0,0 +1,368 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using NuGet.Protocol.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Threading;
+
+namespace GVFS.Common
+{
+    public class NuGetUpgrader : IProductUpgrader
+    {
+        private static readonly string GitBinPath = GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath();
+
+        private ITracer tracer;

```

Comment:
VS says this isn't used... it probably should be.

Reply:
Yeah - I have a TODO on my local notes to add more tracing. I added it to the PR Description TODO list as well.